### ISSUE TITLE
setData not calling computed set() function

### ISSUE THREAD

**Issue by Marcellofabrizio** [vuejs/vue-test-utils#1791]
I'm using **v1.1.3**

I have a component that uses an input and a calendar to set a value that I pass as prop using v-model.
The `inputDate` is a computed that has a getter function that returns a formatted string based on the `value`. And has a setter function that receives a new value, creates a new Date object out of it and emits to the `value`.

It would look something like this:

```js
    inputDate: {
      get() {
        if(this.value) {
          // formats the date object and returns a string for the input
        } 
      },

      set(value) {
          
          if(value.length < 10) {
            if(value === "") {
              this.$emit('input', value)
            }
            return
          }

          // will format the string and create a new date
          let newDateValue = this.createDate(value) 

          this.$emit('input', newDateValue)
      }
    },
```
Now, when I try to replicate this behavior on a unit test using setData, I will get no result

```js
it("Test inputDate", async () => {
        const value = null
        const expected = new Date(2021, 1, 17)
        const wrapper = mount(EternoDatePicker, {
            propsDate: {
                value: value
            }
        })

        await wrapper.setData({ inputDate: "17/02/2021"})

        expect(wrapper.vm.value).toEqual(expected)
       /* Expected value to have received the new value 
       *  after the inputDate is setted, but value is 
       */ undefined.
    })
```

It seems to me that the setter function in this case is not being called.

**Comment by afontcu** [comment]
Hi! Can really test it right now – I'm away from my keyboard. But could you check if using `wrapper.setProps({ value: whatever })` yields better results? Also, could you check if `emitted()` actually returns the expected event?

**Comment by Marcellofabrizio** [comment]
I used `wrapper.setProps` and `value` updated. But I used `wrapper.vm.$emit('input', expected)` to update the value and the value didn't update, althoght `emmited()` confirmed that the event was called.

**Comment by Marcellofabrizio** [comment]
Hi. I figured it out today after some debugging. The value I was passing to my computed was updating it. The problem was that I was expecting the value from the parent prop to be updated and emitted back to me, which wouldn't be possible as the parent component is not mounted. I ended up using `emitted` in the wrong way. All I had to do was to use `expect(wrapper.emitted('input')[0]).toEqual([expected])` to check the emitted event list to see if the value was indeed emitted. Anyway, thanks for the help!
